File: Java/HLC04_EjemplosCodigo1_5/02_Ud4_ServicioDescargaSinHilos/app/src/main/java/com/pms/a02_ud4_serviciodescargasinhilos/MainActivity.java
package com.pms.a02_ud4_serviciodescargasinhilos;

import android.content.Intent;
import android.support.v7.app.AppCompatActivity;
import android.os.Bundle;
import android.view.View;

public class MainActivity extends AppCompatActivity {

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_main);
    }
    //onClick() boton 'arrancarServicio'
    public void arrancarServicio(View view) {
        startService(new Intent(getBaseContext(), MyService.class));
    }
    //onClick() botón 'pararServicio'
    public void pararServicio(View view) {
        stopService(new Intent(getBaseContext(), MyService.class));
    }
}

File: Kotlin/HLC04_EjemplosCodigo1_5/02_Ud4_ServicioDescargaSinHilos/app/src/main/java/com/pms/a02_ud4_serviciodescargasinhilos/MainActivity.kt
package com.pms.a02_ud4_serviciodescargasinhilos

import android.content.Intent
import android.os.Bundle
import android.support.v7.app.AppCompatActivity
import android.view.View

class MainActivity : AppCompatActivity() {
    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_main)
    }

    //onClick() boton 'arrancarServicio'
    fun arrancarServicio(view: View?) {
        startService(Intent(baseContext, MyService::class.java))
    }

    //onClick() botón 'pararServicio'
    fun pararServicio(view: View?) {
        stopService(Intent(baseContext, MyService::class.java))
    }
}